src/views/NoticeOfWithdrawal.vue

```diff
@@ -535,7 +543,8 @@ export default class NoticeOfWithdrawal extends Mixins(CommonMixin, DateMixin, F
           } as StaffPaymentIF
         }
 
-        // load POA and arrangement properties
+        // load Court Order, POA and arrangement properties
+        const courtOrderNumber = filing.noticeOfWithdrawal.courtOrderNumber
```

Comment:
Can you post sample filing JSON with all properties present?

Ultimately, I'm wondering if the schema should be updated, or if we should add validations in Legal API.

Ref: https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/notice_of_withdrawal.json

Reply:
{
    "filing": {
        "header": {
            "name": "noticeOfWithdrawal",
            "certifiedBy": "AQ",
            "email": "Aimee.Gao@gov.bc.ca",
            "date": "2024-12-13",
            "documentOptionalEmail": "test@test.ca",
            "routingSlipNumber": "123456789",
            "priority": false
        },
        "business": {
            "foundingDate": "2024-11-15T21:58:25.000+00:00",
            "identifier": "BC0883739",
            "legalName": "SEVERIN CONVERT ULC TO BC LIMITED CORP.",
            "legalType": "BC"
        },
        "noticeOfWithdrawal": {
            "filingToBeWithdrawn": "153004",
            "referenceNumber": "TT123",
            "courtOrderNumber": "TT0010045",
            "hasPlanOfArrangement": true,
            "hasComeIntoEffect": true
        }
    }
}

Reply:
I do see a mismatch between the JSON and the schema:
- missing filingId (this is probably the filing to withdraw)
- court order properties should be in court order sub-object

Maybe also look at what was done in [23349](https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/23349) regarding validation.

Reply:
Also, I don't think that's how we save the PoA state.

Reply:
Got it, I will update for the court order properties including the POA.
For the filingID: is it assigned from LegalServices.createFiling? As this NOW filing is not supported by LegalServices yet, the filingID is not assigned

Reply:
Quick question: if there is a POA, does it mean court order number is necessary?
Like in Consent to Continue Out:
![image](https://github.com/user-attachments/assets/4852c004-47d0-4090-8376-5f06fc4b14f1)
When the user select POA, they must enter a court order number?

Reply:
No, I think filingId is the id of the filing being withdrawn. I'm not certain -- check with the original author of the schema or the developer of the Legal API validations ticket.

Yes, a court order number is required if the POA checkbox is checked.

Reply:
Thank you.
filingId of the filing being withdrawn: I added it here
![image](https://github.com/user-attachments/assets/b3e41e46-8b04-46ac-8ea7-c9d305cbb72b)
I think it needs to be added in the schema.

POA- I will update this section now

Reply:
> I think it needs to be added in the schema.

I think `filingId` in the schema **is** the filing that is being withdrawn.

Reply:
For the Court Order schema:
https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/court_order.json
![image](https://github.com/user-attachments/assets/9807e906-2a50-4014-b7ee-07610ae8e2d7)
Do we need to add a property for the Come into Effect?
![Uploading image.png…]()

I think the  "effectOfOrder" is for the first check box - POA.
This seems new, I saw the second check box for the first time

Reply:
I'm thinking about add a "effectOfArrangement" to the schema:
"effectOfOrder"for the first check box: means there is a POA, same as other filings
"effectOfArrangement" for the second check box: means the POA has come into effect

Reply:
after update:
{
    "header": {
        "name": "noticeOfWithdrawal",
        "certifiedBy": "AQ",
        "email": "Aimee.Gao@gov.bc.ca",
        "date": "2024-12-13",
        "documentOptionalEmail": "test@test.ca",
        "routingSlipNumber": "123415689",
        "priority": false
    },
    "business": {
        "foundingDate": "2024-11-15T21:58:25.000+00:00",
        "identifier": "BC0883739",
        "legalName": "SEVERIN CONVERT ULC TO BC LIMITED CORP.",
        "legalType": "BC"
    },
    "noticeOfWithdrawal": {
        "filingId": "153004",
        "referenceNumber": "TT0001",
        "courtOrder": {
            "fileNumber": "TT0010045",
            "effectOfOrder": "planOfArrangement",
            "effectOfArrangement": "arrangementEffect"
        }
    }
}

Reply:
I think we label the Reference Number `folioNumber` in other filings, and it's actually in the header object.

I think it's allowed to add properties to a schema object -- like you've done for effectOfArrangement -- but I've seen a more recent UI design and this is a normal court order object (that applies to THIS filing). See other filings for examples.

Meanwhile, the 2 checkboxes apply to the FE filing and are separate from the court order object above. So, you need 2 properties for those, not in the court order object.

Maybe you should take a peek at the more recent UI designs just so you know what's (probably) coming and so that you can get the data handling correct now. Ask Jacqueline for a link.

Reply:
> Can you post sample filing JSON with all properties present?
> 
> Ultimately, I'm wondering if the schema should be updated, or if we should add validations in Legal API.
> 
> Ref: https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/notice_of_withdrawal.json

Yes, we may need to update the schema and validations for NoW, since those tickets were done before all the designs were at place.

Reply:
Sample sent to me via chat. Looks good for now.
```
{
   "header": {
       "name": "noticeOfWithdrawal",
       "certifiedBy": "AQ",
       "email": "Aimee.Gao@gov.bc.ca",
       "date": "2024-12-16",
       "documentOptionalEmail": "test@test.ca",
       "bcolAccountNumber": "000000",
       "datNumber": "C1133435",
       "folioNumber": "123",
       "priority": false
   },
   "business": {
       "foundingDate": "2024-11-15T21:58:25.000+00:00",
       "identifier": "BC0883739",
       "legalName": "SEVERIN CONVERT ULC TO BC LIMITED CORP.",
       "legalType": "BC"
   },
   "noticeOfWithdrawal": {
       "filingId": "153004",
       "courtOrder": {
           "fileNumber": "TT0010045",
           "effectOfOrder": "planOfArrangement"
       },
       "planOfArrangement": {
             "partOfPoa": true,
            "comeIntoEffect": true
   }
}
```